In [7]:
import numpy as np
from scipy.stats import t
import scipy
from scipy import stats

In [8]:
help(stats.ttest_1samp)

Help on function ttest_1samp in module scipy.stats._stats_py:

ttest_1samp(
    a,
    popmean,
    axis=0,
    nan_policy='propagate',
    alternative='two-sided',
    *,
    keepdims=False
)
    Calculate the T-test for the mean of ONE group of scores.

    This is a test for the null hypothesis that the expected value
    (mean) of a sample of independent observations `a` is equal to the given
    population mean, `popmean`.

    Parameters
    ----------
    a : array_like
        Sample observations.
    popmean : float or array_like
        Expected value in null hypothesis. If array_like, then its length along
        `axis` must equal 1, and it must otherwise be broadcastable with `a`.
    axis : int or None, default: 0
        If an int, the axis of the input along which to compute the statistic.
        The statistic of each axis-slice (e.g. row) of the input will appear in a
        corresponding element of the output.
        If ``None``, the input will be raveled before co

In [9]:
data = [14.2, 15.1, 13.8, 14.5, 14.8, 15.4, 13.9, 14.6, 14.1, 14.7]
data

[14.2, 15.1, 13.8, 14.5, 14.8, 15.4, 13.9, 14.6, 14.1, 14.7]

STEP 1

H0 : mu = 15 (battery life is the same as it's claimed)

H1 : mu != 15 (the battery life is different than it's claimed)

STEP 2

alpha = 0.05

STEP 3

Calculate test statistic

In [10]:
s_mean = np.mean(data)
s_std = np.std(data, ddof = 1)
se = s_std / np.sqrt(10)
t_stat = (s_mean - 15) / se
s_mean,s_std,se, t_stat

(np.float64(14.51),
 np.float64(0.5173651192984183),
 np.float64(0.16360521589077368),
 np.float64(-2.9950145374774277))

STEP 4

Find crit and p-values

In [11]:
t_crit = stats.t.ppf(1 - 0.05 / 2, 9)
p_value = 2 * (1 - stats.t.cdf(abs(t_stat), 9))
t_crit, p_value

(np.float64(2.2621571628540993), np.float64(0.01507777596795834))

In [13]:
stats.ttest_1samp(data,15,alternative = 'two-sided')

TtestResult(statistic=np.float64(-2.9950145374774277), pvalue=np.float64(0.015077775967958422), df=np.int64(9))

CONCLUSION:

Because the p-value(0.015) is smaller than the alpha(0.05), we can reject the null hypothesis. The t-value is negative, meaning there is significant data to suggest the battery life is less than 15 hours long.